src/optimizer/factorization_rewriter.cpp

```diff
@@ -189,7 +189,8 @@ std::shared_ptr<planner::LogicalOperator> FactorizationRewriter::appendFlattenIf
     if (op->getSchema()->getGroup(groupPos)->isFlat()) {
         return op;
     }
-    auto flatten = std::make_shared<LogicalFlatten>(groupPos, std::move(op));
+    // we don't need cardinality after planning is complete
+    auto flatten = std::make_shared<LogicalFlatten>(groupPos, std::move(op), 0 /* cardinality */);
```

Comment:
@Guodong Jin is this accurate or do we still need the cardinality here?

Reply:
I don't think this is accurate. We should still keep cardinality after this.

Reply:
I'll do this in a separate PR, updated comment to TODO